File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;
import java.io.*;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        HashMap<String, Integer> hm = new HashMap<>();
        
        for(String num : phone_book){
            hm.put(num, 1);
        }
        
        for(int i = 0; i < phone_book.length; i++){
            for(int j = 0; j < phone_book[i].length(); j++){
                String temp = phone_book[i].substring(0,j);
                if(hm.containsKey(temp)){
                    answer = false;
                    break;
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.cpp
#include <string>
#include <vector>
#include <unordered_map>
using namespace std;

bool solution(vector<string> phone_book) {
    bool answer = true;
    unordered_map<string, int> m;
    
    for (auto s : phone_book) {
        m[s] = 1;
    }
    
    for (int i = 0; i < phone_book.size(); i++) {
        for (int j = 0; j < phone_book[i].size(); j++) {
            string temp = phone_book[i].substr(0, j);
            if (m[temp]) return false;
        }
    }
    return answer;
}